src/std/db/postgresql.ss

```diff
@@ -1,97 +1,110 @@
 ;;; -*- Gerbil -*-
-;;; (C) vyzo
+;;; (C) vyzo, drewc
 ;;; PostgreSQL dbi interface
 
 (import :std/db/dbi
         :std/db/postgresql-driver
         :std/iter
         :std/misc/channel
+        :std/misc/list
         :std/srfi/19)
 (export postgresql-connect
+        (struct-out postgresql-command
```

Comment:
do we need to export the structs?

Reply:
I think so. For one thing the `postgresql-command-complete` and the `postgresql-command-notices` are needed. I've personally found that `...-statement?` is very useful as well. 

Beyond that, `postgresql-statement-[cols|params] has the type info which is extremely useful when interacting with postgresql outside of DBI and especially without a catalog. 

Finally, `postgresql-query` is the entire reasoning behind this patch, and exporting it has been very handy as well. In particular the (not in the tests yet) `greedy: #f` query means that we are free to parse whatever comes in on the `-input` however we want but we will be given the `-command` or `-statement` with the freedom to parse the incoming or play with the notifications BEFORE the rest of the query is pumped if the query-limit was reached.

So, in that sense we "need" all the other structs as knowing the cols/params/notices outside of the generic dbi is necessary when doing it that way.

Make sense? I'll show you my [PostgresUI](https://github.com/drewc/PostgresUI) work later on today that uses a lot of that stuff :).

Reply:
ok...